diff --git a/src/search/landmarks/landmark_count_heuristic.cc b/src/search/landmarks/landmark_count_heuristic.cc
index 0a39d9d55..815afa59a 100644
--- a/src/search/landmarks/landmark_count_heuristic.cc
+++ b/src/search/landmarks/landmark_count_heuristic.cc
@@ -22,6 +22,8 @@
 #include <limits>
 #include <unordered_map>
 
+#include "util.h"
+
 using namespace std;
 using utils::ExitCode;
 
@@ -80,6 +82,11 @@ LandmarkCountHeuristic::LandmarkCountHeuristic(const options::Options &opts)
                  << " orderings." << endl;
     lm_status_manager = utils::make_unique_ptr<LandmarkStatusManager>(*lgraph);
 
+
+    if (opts.get<bool>("dump")) {
+        dump_landmark_graph(task_proxy, *lgraph);
+    }
+
     if (admissible) {
         if (opts.get<bool>("optimal")) {
             lm_cost_assignment = utils::make_unique_ptr<LandmarkEfficientOptimalSharedCostAssignment>(
@@ -361,6 +368,7 @@ static shared_ptr<Heuristic> _parse(OptionParser &parser) {
                             "(see OptionCaveats#Using_preferred_operators_"
                             "with_the_lmcount_heuristic)", "false");
     parser.add_option<bool>("alm", "use action landmarks", "true");
+    parser.add_option<bool>("dump", "dump landmark graph", "false");
     lp::add_lp_solver_option_to_parser(parser);
     Heuristic::add_options_to_parser(parser);
     Options opts = parser.parse();
